social-net/frontend/src/components/Header.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import Menu from "./Menu";
+import ButtonAuthPage from '../components/buttons/ButtonAuthPage';
+import ButtonRegPage from '../components/buttons/ButtonRegPage';
+import ButtonOut from '../components/buttons/ButtonOut';
+import ButtonUserPage from '../components/buttons/ButtonUserPage';
+
+class Header extends React.Component {
+    checkAuth = () => {
```

Comment:
"checkAuth" is not a very good name for a function with this behavior. In most cases, "check" means that the function returning boolean. I'd recommend using something like "renderAuth".